### PR TITLE
DDP-4378 api: rename activity name and title

### PR DESCRIPTION
**Pull request by yufengwng**
Renamed activity attributes:
* `name` -> `title`
* `dashboardName` -> `name`

Backend changes are in #79, Angular SDK changes are in https://github.com/broadinstitute/ddp-angular/pull/45

### CODE REVIEW

**Comment by bskinner** [comment]
Is there a critical reason why this needs to be in now, or is this just a cosmetic change?

**Comment by yufengwng** [comment]
It's not critical. I have fixed the underlying issue on the server-side while keeping the API the same. When's a better time to submit this API change, @bskinner?

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 1]
```diff
@@ -10,7 +10,6 @@ ActivityInstanceStatusType:
     - COMPLETE
     - CREATED
     - IN_PROGRESS
-    - TERMINATED
```
Comment (bskinner) [inline pepper-apis/docs/specification/src/components/schemas.yml:13]: This change is not noted in the PR's description- why is this needed at this time?

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 2]
```diff
@@ -114,49 +113,59 @@ Activity:
   required:
     - activityType
     - formType
+    - activityCode
     - guid
+    - title
```
Comment (bskinner) [inline pepper-apis/docs/specification/src/components/schemas.yml:118]: Are these values guaranteed to be present in the response, or are we considering them nullable?

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 3]
```diff
@@ -169,18 +178,21 @@ Activity:
...
   type: object
   title: Activity Summary
   required:
     - activityCode
-    - activityDashboardName
-    - activityDescription
+    - activityType
+    - activitySubtype
     - activityName
+    - activityTitle
```
Comment (bskinner) [inline pepper-apis/docs/specification/src/components/schemas.yml:192]: What is the requirement for this change? This is not cosmetic, and results in a breaking change to the API.

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 4]
```diff
@@ -1023,15 +1040,15 @@ PrequalifierModel:
   type: object
   required:
     - guid
-    - name
+    - title
```
Comment (bskinner) [inline pepper-apis/docs/specification/src/components/schemas.yml:1052]: What is the requirement behind this change? This is a breaking change to the API and will require client modifications.

**Review by bskinner — CHANGES_REQUESTED** [review]
There are non-cosmetic changes to the API which break compatibility with existing clients.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 1 above)
Comment (yufengwng) [reply pepper-apis/docs/specification/src/components/schemas.yml:13]: This status is no longer in the API (it has been removed some time ago). Just cleaning things up.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 2 above)
Comment (yufengwng) [reply pepper-apis/docs/specification/src/components/schemas.yml:118]: `title` is guaranteed and non-null.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 3 above)
Comment (yufengwng) [reply pepper-apis/docs/specification/src/components/schemas.yml:192]: I'm not aware of a product/stakeholder requirement for this change. This was done as part of refactor for DDP-4378, which helps resolve bug DDP-4376.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 4 above)
Comment (yufengwng) [reply pepper-apis/docs/specification/src/components/schemas.yml:1052]: We can postpone API change. When is a better time?

Comment (yufengwng) [reply pepper-apis/docs/specification/src/components/schemas.yml:1052]: Also, this `PrequalifierModel` is used for this "prequalifer" API which I don't think any studies uses. I'm inclined to remove this API completely.

Comment (bskinner) [reply pepper-apis/docs/specification/src/components/schemas.yml:1052]: Let me take a look at both the apps and verify that. If neither one is using it, I'm game for adding a ticket to remove the API. One less vestigial endpoint will be nice.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 1 above)
Comment (bskinner) [reply pepper-apis/docs/specification/src/components/schemas.yml:13]: Let's try and keep unrelated changes to their own PR in the future. Even if it's not being used, there may be client logic & dependencies on this and it does constitute a breaking change.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 2 above)
Comment (bskinner) [reply pepper-apis/docs/specification/src/components/schemas.yml:118]: Looking at [DDP-4378](https://broadinstitute.atlassian.net/browse/DDP-4378), both the `title` and `subtitle` are mentioned to be `optional`, not required, which doesn't seem to mesh with this. Are these the properties being referred to in the ticket?

Comment (yufengwng) [reply pepper-apis/docs/specification/src/components/schemas.yml:118]: They're optional as in study does not have to define it. If `title` is not defined, it will be blank. `subtitle` is nullable.

**Comment by zyme** [comment]
Ping!  This one is on hold...is now a better time to pop it off the queue?

**Comment by bskinner** [comment]
If we can batch up a bunch of API changes together, I'd be great to get this out.

**Comment by yufengwng** [comment]
Recent API changes so far are additive. I think this is the only change that's breaking. I can pull together some doc describing all the changes if that helps?

**Comment by bskinner** [comment]
That would be helpful. I want to get as many of these in at one time as possible since each release requires me to create a new tag for both the apps & SDKs, update the dependencies in each app, and resubmit to the App Stores for review. In the future, expect at least a 2-3 week lead time for any breaking changes.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 1 above)
Comment (yufengwng) [reply pepper-apis/docs/specification/src/components/schemas.yml:13]: Good point. However, I truly believe that this was added in here haphazardly and wasn't cleaned up. The chances of clients depending on this value is UNLIKELY since the sever will not return such a value. I'll like to remove this. This has been noted in DDP-4471. Resolving comment.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 2 above)
Comment (yufengwng) [reply pepper-apis/docs/specification/src/components/schemas.yml:118]: I believe this is addressed. Resolving comment.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 3 above)
Comment (yufengwng) [reply pepper-apis/docs/specification/src/components/schemas.yml:192]: Tracked in DDP-4471. Resolving comment.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 4 above)
Comment (yufengwng) [reply pepper-apis/docs/specification/src/components/schemas.yml:1052]: Took note of this in DDP-4515. Resolving comment.

**Comment by yufengwng** [comment]
FYI I have revived this PR.

**Review by bskinner — APPROVED** [review]
Let's do it! The matching ticket in Jira is labeled as being a breaking change as well, correct?

**Comment by yufengwng** [comment]
please run tests

**Review by fat23cat — APPROVED** [review]

**Comment by yufengwng** [comment]
please run tests
